ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/identity_management_service.proto

```diff
@@ -0,0 +1,138 @@
+  rpc ListKnownParties (ListKnownPartiesRequest) returns
+    (ListKnownPartiesResponse);
+
+  // Adds a new party to the set managed by the backing participant.
+  // Caller specifies a party identifier suggestion, the actual identifier
+  // allocated might be different and is implementation specific.
+  // daml-on-sql: suggestion's uniqueness is checked and call rejected if the
+  // identifier is already present
+  // daml-on-sawtooth: suggestion is prefixed with backing participant
+  // identifier and the composite is tested for uniqueness
+  // canton: completely different globally unique identifier is allocated.
+  // Behind the scenes calls to an internal protocol are made. As that protocol
+  // is richer than the the surface protocol, the arguments take implicit values
+  // Future extension: augment the request with additional meta-data containing
+  // human readable party name, privileges, organization etc.
+  rpc AllocateParty (AllocatePartyRequest) returns (AllocatePartyResponse);
+}
+
+message GetParticipantIdRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message GetParticipantIdResponse {
+  ParticipantDetails participant_details = 1;
+}
+
+message ParticipantDetails {
+
+  string participant_id = 1;
+}
+
+message ListKnownParticipantsRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListKnownParticipantsResponse {
+  repeated ParticipantDetails participant_details = 1;
+}
+
+message ListKnownPartiesRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListKnownPartiesResponse {
+
+  // The details of all DAML parties hosted by the participant.
+  // Required
+  repeated PartyDetails party_details = 1;
+}
+
+message PartyDetails {
+
+  // The ID of a DAML party hosted by the participant.
+  // Required
+  string party_id = 1;
+
+  // Future extension: include metadata fields. There will be two categories
+  // of such fields: common and implementation specific.
+  // CommonMetadata common_metadata = 2;
+  // bytes implementation_metadata = 3;
+  
+  // Canton-specific meta-data would contain the following:
+  // string domain = 2;
+  // string participant = 3;
+  // com.digitalasset.canton.identity.v0.ParticipantPrivilege privilege = 4;
+  // com.digitalasset.canton.identity.v0.TrustLevel trustLevel = 5;
+}
+
+message AllocatePartyRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+
+  // Party to be added to the participant.
+  // Required
+  string party_id_suggestion = 2;
+}
+
+message AllocatePartyResponse {
+
+  // Actual participant id assigned by the backing participant
+  // Required
+  string party_id = 2;
```

Comment:
rename to `allocated_party_id`

Reply:
changed to "party" for consistency reasons